fix: issues with formio/js, formio/protected-eval and a csp header wi…

YourivHDenHaag opened the pull request:
…thout unsafe-eval + added basic working csp header

- Fixes issues with Formio combined with a CSP header without unsafe-eval.

YourivHDenHaag committed:
fix: issues with formio/js, formio/protected-eval and a csp header without unsafe-eval + added basic working csp header

Borai committed:
Merge branch 'next-minor' into feature/fix_formio_csp_issues

YourivHDenHaag committed:
Merge branch 'next-minor' into feature/fix_formio_csp_issues

nielserkamp-ritense commented:
I see a customization of the FromIO protected eval. And a change to the default csp headers. But I prefer to not change those, for the simple reason that they would not pass an audit for DigiD.

YourivHDenHaag committed:
fix: use formio's newest evaluator and interpolate fixes

YourivHDenHaag committed:
revert: csp change

YourivHDenHaag commented:
@nielserkamp-ritense Formio has fixed the evaluator and interpolation issues a few days ago, so i replaced my fix with the new fixes from Formio. 

I understand that the CSP header is not DigiD audit compliant, but the original header does not function correctly.
<img width="1483" height="134" alt="image" src="https://github.com/user-attachments/assets/90bcc6de-2ceb-471e-9f7f-8268f7b8f18b" />
I reverted the CSP change, however, be aware that the current CSP configuration does nothing.

anna-ritense commented:
> current CSP configuration
what is this "current " configuration? the libraries dont set any by default. It's up to the implementation to configure CSP in their own implementation.

anna-ritense commented:
> be aware that the current CSP configuration does nothing.

we are aware that it does nothing, since this app is only for developing the libraries. every implementation can implement their own portal from the libraries and thus also configure their own CSP.
maybe we should remove that meta tag to avoid confusion?

but evaluator issues being fixed by formio is good news indeed! 👍

YourivHDenHaag commented:
> > be aware that the current CSP configuration does nothing.
> 
> we are aware that it does nothing, since this app is only for developing the libraries. every implementation can implement their own portal from the libraries and thus also configure their own CSP. maybe we should remove that meta tag to avoid confusion?
> 
> but evaluator issues being fixed by formio is good news indeed! 👍

I understand, removing the meta tag to prevent confusion sounds reasonable. But maybe is adding a basic one which blocks unsafe evaluations not a bad idea either, so we can at least check if the nl-portal-frontend-libraries works without unsafe eval expressions.

But I’ll leave that choice up to you @nielserkamp-ritense @anna-ritense.
